content/courses/advanced-ada/chapters/types.rst

```diff
@@ -445,10 +445,125 @@ Mutually dependent types
 Type view
 ---------
 
-.. todo::
+Ada distinguishes between the partial and the full view of a type. The full
+view is a type declaration that contains all the information needed by the
+compiler. For example, the following declaration of type :ada:`R` represents
+the full view of this type:
 
-    Complete section!
+.. code:: ada compile_button project=Courses.Advanced_Ada.Types.Full_View
+
+    package Full_View is
+
+       --  Full view of the R type:
+       type R is record
+          I : Integer;
+       end record;
+
+    end Full_View;
+
+As soon as we start applying encapsulation and information hiding |mdash| via
+the :ada:`private` keyword |mdash| to a specific type, we are introducing a
+partial view and making only that view compile-time visible to clients. Doing
+so requires us to introduce the private part of the package (unless already
+present). For example:
+
+.. code:: ada compile_button project=Courses.Advanced_Ada.Types.Partial_Full_View
+
+    package Partial_Full_Views is
+
+       --  Partial view of the R type:
+       type R is private;
+
+    private
+
+       --  Full view of the R type:
+       type R is record
+          I : Integer;
+       end record;
+
+    end Partial_Full_Views;
+
+As indicated in the example, the :ada:`type R is private` declaration is the
+partial view of the :ada:`R` type, while the :ada:`type R is record [...]`
+declaration in the private part of the package is the full view.
+
+Although the partial view doesn't contain the full type declaration, it
+contains very important information for the users of the package where it's
+declared. In fact, the partial view of a private type is all that users
+actually need to know to effectively use this type, while the full view is only
+needed by the compiler.
+
+In the previous example, the partial view indicates that :ada:`R` is a private
+type, which means that, even though users cannot directly access any
+information stored in this type |mdash| for example, read the value of the
+:ada:`I` component of :ada:`R` |mdash|, they can use the :ada:`R` type to
+declare objects. For example:
+
+.. code:: ada compile_button project=Courses.Advanced_Ada.Types.Partial_Full_View
+
+    with Partial_Full_Views; use Partial_Full_Views;
+
+    procedure Main is
+       --  Partial view of R indicates that R exists as a private type,
+       --  so we can declare objects of this type:
+       C : R;
+    begin
+       --  But we cannot directly access any information declared in the full
+       --  view of R:
+       --
+       --  C.I := 42;
+       --
+       null;
```

Comment:
Maybe should have a comment on the line doing the assignment, saying it is illegal